lib/core/src/lib/datatable/components/datatable/datatable.component.ts

```diff
@@ -527,9 +544,14 @@ export class DataTableComponent implements OnInit, AfterContentInit, OnChanges,
     private handleRowSelection(row: DataRow, e: KeyboardEvent | MouseEvent) {
         if (this.data) {
             if (this.isSingleSelectionMode()) {
-                this.resetSelection();
-                this.selectRow(row, true);
-                this.emitRowSelectionEvent('row-select', row);
+                if (row.isSelected) {
+                    this.resetSelection();
```

Comment:
You use it in both if branches, can be moved outside if statement

Reply:
actually method resetSelection() is causing side effects to isSelected property, so either it requires assaign to const isRowSelected or calling it twice

Reply:
Okay, fine for me